pkg/cmd/experimental/registry/registry.go

```diff
@@ -0,0 +1,267 @@
+package registry
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	kclientcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	configcmd "github.com/openshift/origin/pkg/config/cmd"
+	dapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/generate/app"
+)
+
+const longDesc = `
+Install or configure a Docker registry for OpenShift
+
+This command helps to setup a Docker registry integrated with OpenShift to provide
+notifications when images are pushed. With no arguments, the command will check for the existing registry
+service called 'docker-registry' and perform some diagnostics to ensure the registry is properly
+configured and functioning.
+
+If a registry service does not exist, the --create flag can be passed to
+create a deployment configuration and service that will run the registry. If you are
+running your registry in production, you should pass --replicas=2 or higher to ensure
```

Comment:
does replica 2 really work correctly?  What happens when a docker push and it gets loadbalanced to one repo, and then i docker pull it the pull gets loadbalanced to the other repo, given that we don't have shared volumes?